core/src/main/java/org/elasticsearch/monitor/jvm/JvmStats.java

```diff
@@ -79,14 +79,15 @@ public static JvmStats jvmStats() {
                 MemoryUsage usage = memoryPoolMXBean.getUsage();
                 MemoryUsage peakUsage = memoryPoolMXBean.getPeakUsage();
                 String name = GcNames.getByMemoryPoolName(memoryPoolMXBean.getName(), null);
-                if (name == null) { // if we can't resolve it, its not interesting.... (Per Gen, Code Cache)
-                    continue;
+                if (name == null) { // if we can't resolve it, its not GC related.... (Metaspace, Code Cache)
+                    name = memoryPoolMXBean.getName();
                 }
+                // MemoryUsage.getMax() can be -1, indicating the maximum is undefined (e.g. metaspace max)
                 pools.add(new MemoryPool(name,
-                        usage.getUsed() < 0 ? 0 : usage.getUsed(),
-                        usage.getMax() < 0 ? 0 : usage.getMax(),
-                        peakUsage.getUsed() < 0 ? 0 : peakUsage.getUsed(),
-                        peakUsage.getMax() < 0 ? 0 : peakUsage.getMax()
+                        usage.getUsed(),
+                        usage.getMax() == -1 ? Long.MAX_VALUE : usage.getMax(),
```

Comment:
I looked into this change, I don't think its reasonable to try to fold into this. The problem is that its hard to keep things as `-1` because `ByteSizeValue` is the api around all this, and its unclear that it supports negative values: there is no guard to prevent it, but at least its serialization would not work (uses vlong), who knows about its formatting and other stuff :)

Perhaps its possible to change the API to return null to indicate the unbounded case, but that's a lot of work if we want to apply it across the whole stats api too. Definitely requires simple serialization tests, which i started writing, but are missing everywhere, and there are bugs that reflect that, e.g. classloader statistics just get dropped on the floor.

Basically, I see a lot of bugs everywhere, and no clear path to fixing the situation without a lot of work. For now, I guess we can just remain in the dark on these statistics.